### PR TITLE
Refactor project structure

### PR DESCRIPTION
**Pull request by 2m**
This PR updates and refactors project structure to:

* be a single project sbt build
* use sbt-scalafmt
* use Travis stages for testing and deployment
* publish every master commit to Maven Central with a version derived from `git describe`

### CODE REVIEW

File: src/main/scala/akka/stream/contrib/UnfoldFlow.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+package akka.stream.contrib
+
+import akka.stream.{Attributes, FanOutShape2}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import akka.util.Timeout
+
+/** INTERNAL API */
+private[akka] abstract class UnfoldFlowGraphStageLogic[O, S, E] private[stream] (shape: FanOutShape2[O, S, E],
+                                                                                 seed: S,
+                                                                                 timeout: Timeout)
```
Comment (raboof) [inline src/main/scala/akka/stream/contrib/UnfoldFlow.scala]: Wow, such indent... can we move things to a new line or does scalafmt not allow that?

Comment (2m) [reply src/main/scala/akka/stream/contrib/UnfoldFlow.scala]: I used the same formatting rules as in Alpakka and do not want to diverge from those.

File: build.sbt [hunk 2]
```diff
@@ -1,6 +1,66 @@
...
+  "com.typesafe.akka" %% "akka-stream-testkit" % AkkaVersion % Test,
+  "junit" % "junit" % "4.12" % Test, // Common Public License 1.0
+  "com.novocode" % "junit-interface" % "0.11" % Test, // BSD-like
+  "com.google.jimfs" % "jimfs" % "1.1" % Test, // ApacheV2
+  "org.scalatest" %% "scalatest" % "3.0.6" % Test, // ApacheV2
+  "com.miguno.akka" %% "akka-mock-scheduler" % "0.5.4" % Test // ApacheV2
+)
+
+organizationName := "Lightbend Inc."
+organizationHomepage := Some(url("http://www.lightbend.com"))
```
Comment (ennru) [inline build.sbt]:
```suggestion
organizationHomepage := Some(url("https://www.lightbend.com"))
```

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,5 +1,5 @@
-addSbtPlugin("de.heikoseeberger" % "sbt-header"      % "5.0.0")
-addSbtPlugin("org.scalariform"   % "sbt-scalariform" % "1.8.2")
-addSbtPlugin("com.jsuereth"      % "sbt-pgp"         % "1.1.1")
-addSbtPlugin("com.dwijnand"      % "sbt-travisci"    % "1.1.0")
-addSbtPlugin("com.dwijnand"      % "sbt-dynver"      % "3.0.0")
+addSbtPlugin("de.heikoseeberger" % "sbt-header" % "5.1.0")
```
Comment (ennru) [inline project/plugins.sbt]:
```suggestion
addSbtPlugin("de.heikoseeberger" % "sbt-header" % "5.2.0")
```

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.1.2")
+addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "2.4")
+addSbtPlugin("com.dwijnand" % "sbt-dynver" % "3.3.0")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.0.0-RC5")
```
Comment (ennru) [inline project/plugins.sbt]:
```suggestion
addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.0.0")
```

**Pull request merged**